### PR TITLE
Add Bemini

### PR DESCRIPTION
**Pull request by ianklug**
## Description

Add config and keymaps for Bemini keyboard controller

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ianklug/bemini/config.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#define DIRECT_PINS { \
+    { C7, C6, B6, B5, B4, D7, D6, D4, D3, D2, D1, D0, F5 } \
+}
+#define UNUSED_PINS
+
+/* encoder pins */
+#define ENCODERS_PAD_A { F7 }
+#define ENCODERS_PAD_B { F6 }
+/* encoder hardware resolution */
+#define ENCODER_RESOLUTION 2
```
Comment (drashna) [inline keyboards/ianklug/bemini/config.h:41]: This is something that should (usually) be set by users, not globally. 
```suggestion
```

File: keyboards/ianklug/bemini/config.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+/* gamepad buttons and axes */
+#define JOYSTICK_BUTTON_COUNT 16
+#define JOYSTICK_AXES_COUNT 2
+
+/* n-key rollover */
+#define FORCE_NKRO
+
+/* 1000hz polling */
+#define USB_POLLING_INTERVAL_MS 1
+#define QMK_KEYS_PER_SCAN 12
```
Comment (drashna) [inline keyboards/ianklug/bemini/config.h:55]: There should be no need for this.  Default polling rate is already set to this. 

```suggestionz
```

File: keyboards/ianklug/bemini/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        if (clockwise) {
+            tap_code(KC_LSFT);
+        } else {
+            tap_code(KC_LCTL);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/ianklug/bemini/keymaps/default/keymap.c:59]:
```suggestion
```

File: keyboards/ianklug/bemini/keymaps/default_p2/keymap.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {
+    if (clockwise) {
+      tap_code(KC_RSFT);
+    } else {
+      tap_code(KC_RCTL);
+    }
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/ianklug/bemini/keymaps/default_p2/keymap.c:59]:
```suggestion
```

File: keyboards/ianklug/bemini/bemini.c [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "bemini.h"
+
```
Comment (drashna) [inline keyboards/ianklug/bemini/bemini.c:18]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code(KC_LSFT);
        } else {
            tap_code(KC_LCTL);
        }
    }
    return true;
}
#endif
```

File: keyboards/ianklug/bemini/rules.mk [hunk 6]
```diff
@@ -0,0 +1,8 @@
+MCU = atmega32u4
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/ianklug/bemini/rules.mk:7]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by ianklug** [comment]
your suggestions have been implemented, thanks drashna!

**Review by drashna — APPROVED** [review]

File: keyboards/ianklug/bemini/info.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+    "keyboard_name": "bemini",
+    "url": "https://github.com/ianklug/bemini",
+    "maintainer": "ianklug",
+    "width": 10,
+    "height": 3,
```
Comment (fauxpark) [inline keyboards/ianklug/bemini/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by ianklug** [comment]
closed - reworking for new hardware MCU

**Pull request closed without merging**